hazelcast/src/main/java/com/hazelcast/multimap/impl/MultiMapContainer.java

```diff
@@ -214,4 +221,64 @@ public long getLockedCount() {
     public ObjectNamespace getObjectNamespace() {
         return objectNamespace;
     }
+
+    @Override
+    public MultiMapValue merge(SplitBrainMergeEntryView<Data, MultiMapMergeContainer> mergingEntry,
+                               SplitBrainMergePolicy mergePolicy) {
+        SerializationService serializationService = nodeEngine.getSerializationService();
+        mergePolicy.setSerializationService(serializationService);
+
+        Data key = mergingEntry.getKey();
+        MultiMapMergeContainer mergingContainer = mergingEntry.getValue();
+        MultiMapValue existingValue = getMultiMapValueOrNull(key);
+
+        if (existingValue == null) {
+            return mergeNewValue(mergePolicy, serializationService, key, mergingContainer);
+        }
+        return mergeExistingValue(mergePolicy, serializationService, key, mergingContainer, existingValue);
+    }
+
+    private MultiMapValue mergeNewValue(SplitBrainMergePolicy mergePolicy, SerializationService ss, Data key,
+                                        MultiMapMergeContainer mergingContainer) {
+        boolean isBinary = getConfig().isBinary();
+        MultiMapValue mergedValue = null;
+        for (MultiMapRecord mergeRecord : mergingContainer.getRecords()) {
+            SplitBrainMergeEntryView<Data, Object> mergeEntry = createSplitBrainMergeEntryView(mergingContainer, mergeRecord);
```

Comment:
Not introduced in this PR but `SplitBrainMergeEntryView` seems a long name to me, i think `merge` already implies healing after split-brain. So `MergingEntryView` can be an alternative name.

Reply:
I used that name so all split-brain related classes are grouped together in the SPI folder. Let's have a look at the naming after all implementation PRs have been merged though. Maybe we can improve this.